enforcer-parent/enforcer/src/main/java/org/wso2/choreo/connect/enforcer/security/mtls/MTLSAuthenticator.java

```diff
@@ -0,0 +1,167 @@
+ */
+public class MTLSAuthenticator implements Authenticator {
+    private static final Logger log = LogManager.getLogger(MTLSAuthenticator.class);
+    private final boolean isEnableClientValidation = FilterUtils.getClientValidationStatus();
+    private final boolean isClientCertificateEncode = FilterUtils.getCertificateEncodeStatus();
+
+    @Override
+    public boolean canAuthenticate(RequestContext requestContext) {
+        String cert = "";
+
+        if (isEnableClientValidation) {
+            cert = requestContext.getClientCertificate();
+        } else if (requestContext.getHeaders().containsKey(FilterUtils.getCertificateHeaderName())) {
+            cert = requestContext.getHeaders().get(FilterUtils.getCertificateHeaderName());
+        }
+        if (cert.isEmpty()) {
+            log.debug("Could not find a valid client certificate in the request: " + requestContext.getRequestPath());
+            return false;
+        }
+        return true;
+    }
+
+    @Override
+    public AuthenticationContext authenticate(RequestContext requestContext) throws APISecurityException {
+        TracingTracer tracer;
+        TracingSpan mtlsApiAuthenticatorSpan = null;
+        Scope mtlsApiAuthenticatorSpanScope = null;
+
+        try {
+            if (Utils.tracingEnabled()) {
+                tracer = Utils.getGlobalTracer();
+                mtlsApiAuthenticatorSpan = Utils.startSpan(TracingConstants.MTLS_API_AUTHENTICATOR_SPAN, tracer);
+                mtlsApiAuthenticatorSpanScope = mtlsApiAuthenticatorSpan.getSpan().makeCurrent();
+                Utils.setTag(mtlsApiAuthenticatorSpan, APIConstants.LOG_TRACE_ID,
+                        ThreadContext.get(APIConstants.LOG_TRACE_ID));
+            }
+
+            AuthenticationContext authenticationContext = new AuthenticationContext();
+            KeyStore trustStore = requestContext.getMatchedAPI().getTrustStore();
+            boolean authenticated = false;
+
+            try {
+                X509Certificate clientCert = getClientCertificate(requestContext);
+                String clientCertificateAlias = MtlsUtils
+                        .getMatchedCertificateAliasFromTrustStore(clientCert, trustStore);
+                if (clientCertificateAlias == null) {
+                    log.debug(String.format("Provided client certificate in request: %s is not in the truststore of " +
+                            "the API: %s", requestContext.getRequestPath(), requestContext.getMatchedAPI().getName()));
+                    clientCert = null;
+                }
+                if (clientCert != null) {
+                    authenticated = true;
+                    String clientCertificateTier = requestContext.getMatchedAPI().getMtlsCertificateTiers()
+                            .get(clientCertificateAlias);
+                    if (clientCertificateTier != null && !clientCertificateTier.equals("")) {
+                        authenticationContext.setTier(clientCertificateTier);
```

Comment:
1. Let's check `.getMtlsCertificateTiers().containsKey()` first.
2. Let's switch to `!"".equals(clientCertificateTier)`

Reply:
for 2. we also can directly use `StringUtils.isBlank()`. https://commons.apache.org/proper/commons-lang/javadocs/api-2.6/org/apache/commons/lang/StringUtils.html#isBlank%28java.lang.String%29